tests/fake/test_fake_search.py

```diff
@@ -430,3 +437,29 @@ def test_search_mapped_field_less_than():
     found = client.search_distributor(crit).result().data
 
     assert found == [dist1]
+
+
+def test_search_task():
+    controller = FakeController()
+
+    task1 = Task(
+        id="task1",
+        completed=True,
+        succeeded=True,
+        tags=[
+            "pulp:repository:repo1",
+            "pulp:action:publish",
+        ],
+    )
+
+    controller.insert_task(task1)
```

Comment:
Could you please make this test also insert some tasks which should NOT be found?

Right now it's a weak test, because e.g. if the fake client search were implemented to ignore the criteria and just return every task every time, the test would still pass.

Reply:
Agree. Adding a new task here.